src/units/http_server.rs

```diff
@@ -967,6 +969,104 @@ impl HttpServer {
 
         Json(Ok(()))
     }
+
+    async fn status_keys(State(state): State<Arc<HttpServerState>>) -> Json<KeyStatusResult> {
+        #[derive(Deserialize)]
+        struct KeySetState {
+            keyset: KeySet,
+        }
+
+        #[derive(Deserialize)]
+        struct KeySetConfig {
+            ksk_validity: Option<Duration>,
+            zsk_validity: Option<Duration>,
+            csk_validity: Option<Duration>,
+            autoremove: bool,
+        }
+
+        let state = state.center.state.lock().unwrap();
+
+        let keys_dir = &state.config.keys_dir;
+
+        let now = SystemTime::now()
+            .duration_since(SystemTime::UNIX_EPOCH)
+            .expect("System time is expected to be after UNIX_EPOCH");
+
+        let mut zones = Vec::new();
+        let mut expirations = Vec::new();
+
+        for zone in state.zones.iter() {
+            let mut zone_keys = Vec::new();
+
+            let cfg_path = keys_dir.join(format!("{}.cfg", zone.0.name.to_string().to_lowercase()));
```

Comment:
Maybe use `mk_dnst_keyset_cfg_file_path()` here to avoid hard-coding everywhere the naming template for the file?